test/integration/scheduler/scheduler_test.go

```diff
@@ -305,6 +305,55 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			})
 		})
 
+		ginkgo.It("Should admit workloads when resources are dynamically reclaimed", func() {
+			firstWl := testing.MakeWorkload("first-wl", ns.Name).Queue(prodQueue.Name).
+				PodSets(
+					*testing.MakePodSet("first", 1).Request(corev1.ResourceCPU, "3").Obj(),
+					*testing.MakePodSet("second", 1).Request(corev1.ResourceCPU, "1").Obj(),
+					*testing.MakePodSet("third", 1).Request(corev1.ResourceCPU, "1").Obj(),
+				).
+				Obj()
+			ginkgo.By("Creating first workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, firstWl)).Should(gomega.Succeed())
+
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, firstWl)
+				util.ExpectPendingWorkloadsMetric(prodClusterQ, 0, 0)
+				util.ExpectReservingActiveWorkloadsMetric(prodClusterQ, 1)
+			})
+
+			ginkgo.By("Reclaim one pod from the first workload", func() {
+				gomega.Expect(workload.UpdateReclaimablePods(ctx, k8sClient, firstWl, []kueue.ReclaimablePod{{Name: "second", Count: 1}})).To(gomega.Succeed())
+
+				util.ExpectPendingWorkloadsMetric(prodClusterQ, 0, 0)
+				util.ExpectReservingActiveWorkloadsMetric(prodClusterQ, 1)
+			})
+
+			secondWl := testing.MakeWorkload("second-wl", ns.Name).Queue(prodQueue.Name).
+				PodSets(
+					*testing.MakePodSet("first", 1).Request(corev1.ResourceCPU, "3").Obj(),
+					*testing.MakePodSet("second", 1).Request(corev1.ResourceCPU, "1").Obj(),
+					*testing.MakePodSet("third", 1).Request(corev1.ResourceCPU, "1").Obj(),
+				).
+				Priority(100).
+				Obj()
+			ginkgo.By("Creating the second workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, secondWl)).Should(gomega.Succeed())
+
+				//util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, secondWl)
+				util.ExpectWorkloadsToBePending(ctx, k8sClient, secondWl)
+				util.ExpectPendingWorkloadsMetric(prodClusterQ, 0, 1)
+				util.ExpectReservingActiveWorkloadsMetric(prodClusterQ, 1)
+			})
+
+			ginkgo.By("Reclaim two pods from the second workload", func() {
```

Comment:
```suggestion
			ginkgo.By("Reclaim two pods from the second workload so that the first workload is resumed", func() {
```
To make a connection with the Issue description.